unipop-elastic/src/org/unipop/elastic/common/QueryIterator.java

```diff
@@ -1,65 +1,96 @@
 package org.unipop.elastic.common;
 
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.client.http.JestHttpClient;
+import io.searchbox.core.Search;
+import io.searchbox.core.SearchScroll;
+import io.searchbox.params.Parameters;
 import org.apache.tinkerpop.gremlin.structure.Element;
 import org.elasticsearch.action.search.SearchRequestBuilder;
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.client.Client;
 import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.search.SearchHit;
-import org.unipop.query.StepDescriptor;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.internal.InternalSearchHit;
+import org.jooq.lambda.Seq;
 
+import java.io.IOException;
 import java.util.Iterator;
+import java.util.Map;
 
 public class QueryIterator<E extends Element> implements Iterator<E> {
 
-    private final StepDescriptor stepDescriptor;
-    private SearchResponse scrollResponse;
+    private JestResult scrollResponse;
+    private String scrollId;
     private final Parser<E> parser;
     private final int scrollSize;
-    private Client client;
-    private Iterator<SearchHit> hits;
+    private JestClient client;
+    private Iterator<JsonElement> hits;
+    private final Gson gson;
 
-    public QueryIterator(QueryBuilder query, StepDescriptor stepDescriptor, int scrollSize, int maxSize, Client client,
-                         Parser<E> parser, String... indices) {
-        this.stepDescriptor = stepDescriptor;
+    public QueryIterator(QueryBuilder query, int scrollSize, int maxSize, JestClient client,
+                         Parser<E> parser, String... indices) throws IOException {
         this.scrollSize = scrollSize;
         this.client = client;
         this.parser = parser;
 
-        SearchRequestBuilder searchRequestBuilder = client.prepareSearch(indices).setQuery(query);
+        this.gson = new Gson();
```

Comment:
Is Gson better then the current JSON library we're using? Maybe open an issue for the future to standardize Unipop to a single library...